src/lib.rs

```diff
@@ -702,14 +703,28 @@ pub fn delegate(tokens: TokenStream) -> TokenStream {
     let block: DelegationBlock = syn::parse_macro_input!(tokens);
     let sections = block.segments.iter().map(|delegator| {
         let delegator_attribute = &delegator.delegator;
-
         let functions = delegator.methods.iter().map(|method| {
             let input = &method.method;
-            let signature = &input.sig;
-
+            let mut signature = input.sig.clone();
+            if let syn::Expr::Closure(closure) = delegator_attribute {
+                let additional_input: Vec<FnArg> = closure
+                    .inputs
+                    .iter()
+                    .map(|input| {
+                        if let syn::Pat::Type(pat_type) = input {
+                            syn::parse_quote!(#pat_type)
+                        } else {
+                            unreachable!()
```

Comment:
It would be nice to provide a better error message here, rather than just using `unreachable!`.
At least something like this:
```rust
panic!(
    "Use a type pattern (`a: u32`) for delegation closure arguments"
);
```